test/test_utils.py

```diff
@@ -1860,8 +1873,15 @@ def test_get_element_text_and_html_by_tag(self):
         self.assertEqual(
             get_element_text_and_html_by_tag('span', html),
             (self.GET_ELEMENT_BY_TAG_RES_INNERSPAN_TEXT, self.GET_ELEMENT_BY_TAG_RES_INNERSPAN_HTML))
+
         self.assertRaises(compat_HTMLParseError, get_element_text_and_html_by_tag, 'article', html)
 
+        html = self.GET_ELEMENT_BY_TAG_TEST_STRING_UPPERCASE
+
+        self.assertEqual(
+            get_element_text_and_html_by_tag('SPAN', html),
+            ('nice', html.strip()), html)
```

Comment:
this test seems to demonstrate a misunderstanding of the issue that needs to be solved; we need `get_element_text_and_html_by_tag('span', html)` (lowercase `span`) to return the text and html of an uppercased `<SPAN>` tag

Reply:
Now added a commit which does this.